content/tutorials/mqtt-snake.textile

```diff
@@ -0,0 +1,539 @@
+        }
+      });
+    </script>
+  </body>
+  </html>
+```
+
+This will simply subscribe to the @input@ channel you're publishing to from the controller, and print it into the console. The @TextDecoder@ is required due to MQTT being a binary protocol with no encoding. This means the payload you receive is just raw bits that need to be interpreted. In this case we're interpreting it as text with the @TextDecoder@.
+
+blang[javascript].
+  Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @node controller.js@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+  <a href="/images/tutorials/mqtt-snake-keys-received.png" target="_blank">
+    <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+  </a>
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-js-step7
+
+blang[go].
+  Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @go run controller.go@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+  <a href="/images/tutorials/mqtt-snake-keys-received.png" target="_blank">
+    <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+  </a>
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-go-step7
+
+blang[python].
+  Open up @snake.html@ in a browser, open up the browser's "console":http://wickedlysmart.com/hfjsconsole/, and run your controller by typing @python ./controller.py@ into your command line. Now when you press the arrow keys from your controller, you should see messages being received by the browser in the console.
+
+  <a href="/images/tutorials/mqtt-snake-keys-received.png" target="_blank">
+    <img src="/images/tutorials/mqtt-snake-keys-received.png" style="width: 100%" alt="Sending messages through Ably with MQTT">
+  </a>
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-python-step7
+
+h2. Step 8 - Play Snake!
+
+blang[javascript].
+  With the setup complete, you can now add in the code for the actual snake game. Replace the contents of @snake.html@ with "this":https://github.com/ably/tutorials/blob/mqtt-snake-js-step8/mqtt-snake/snake.html code. Replace REPLACE_WITH_YOUR_API_KEY with your API Key. When you reload the @snake.html@ page you should see your basic Snake game, based off of "this":https://gist.github.com/atimb/1449250 code. Run the controller as before, and you can now play! Use the arrow keys to move, press the space key to pause/play, and try to grow the Snake as large as you can.
```

Comment:
Perhaps make `REPLACE_WITH_YOUR_API_KEY` code formatted i.e. `@REPLACE_WITH_YOUR_API_JEY@`

Reply:
Given the Javascript, Go and Python text is exactly the same with the exception of `"this":https://github.com/ably/tutorials/blob/mqtt-snake-js-step8/mqtt-snake/snake.html`, why not use `<span lang>` blocks. See https://docs.ably.io/client-lib-development-guide/documentation-formatting-guide/#language-specific-content, go to "In-line content with"

Reply:
Please make your links contain meaningful text in all cases.  You have a link "this", which could instead be "our Snake HTML code".  Please review other similar links with misleading links